src/library_wasmfs_jsimpl.js

```diff
@@ -9,6 +9,8 @@ mergeInto(LibraryManager.library, {
   // the JS code that implements them. This is the JS side of the JSImpl* class
   // in C++, together with the js_impl calls defined right after it.
   $wasmFS$backends: {},
+  // Contains a mapping of backend files to their file names.
+  $wasmFS$backendFileNames: {},
```

Comment:
Would it be possible to put any new state in `library_wasmfs_fetch.js` instead?

Reply:
Yes. Done.
Alternatively, I can call `getRelativeFileName()` directly in `getFile()` without caching.